EfficientNet mypy fix

masadcv opened the pull request:
Fixes #2125 .

### Description
Fixes mypy error on some platforms occuring in EfficientNet BlockArgs definition.

### Status
**Ready**

### Types of changes
<!--- Put an `x` in all the boxes that apply, and remove the not applicable items -->
- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests`.

masadcv commented:
@wyli Apolpgies for messing this up, it was supposed to be an easy fix. I cannot seem to figure out how to correct DCO issue here.

wyli commented on monai/networks/nets/efficientnet.py:
```diff
@@ -785,18 +799,6 @@ def _decode_block_list(string_list: List[str]):
-        id_skip: bool
-        se_ratio: Optional[float] = None
 
     def _decode_block_string(block_string: str):
```
looks like the `_decode_block_string` should be a part of `BlockArgs`, and we could add a `to_string` method for completeness?

```py
class BlockArgs(NamedTuple):
    ...
    @staticmethod
    def from_string(inst_string):
        ...
        return a blockargs instance

    def to_string(self):
        ...
        # convert args into the string notation

# usage:
blocks_args = [BlockArgs.from_string(s) for s in string_list]
strings = [b.to_string() for b in block_args]
```

masadcv replied:
@wyli yes, it can be done like that. Let me correct it that way. The reason I left this mechanism of decoding model definitions from a string was to make it future proof where in future we can have support for new EfficientNet model definitions learned through AutoML. These are usually defined as list of strings that can be decoded to make the model.

wyli replied:
exactly, it's a nice idea, let's revisit this when we make the generic AutoML-related APIs

wyli commented:
> @wyli Apolpgies for messing this up, it was supposed to be an easy fix. I cannot seem to figure out how to correct DCO issue here.

no worries perhaps do a "git reset @~2" and commit again with -s, then push with -f

masadcv commented:
@wyli many thanks for help with git instructions, I fixed the commit and added a BlockArgs class with `from_string` and `to_string` functions. I also added a new unittest to test the new functions. This is ready for review, all unittests pass locally.

wyli reviewed (approved):
thanks!
